app/questionnaire/questionnaire_schema.py

```diff
@@ -14,75 +14,96 @@ def __init__(self, questionnaire_json, language_code=DEFAULT_LANGUAGE_CODE):
+    def section(self, section_id: str = None, block_id: str = None):
+        if section_id:
+            return self._sections_by_id.get(section_id)
+
+        if block_id:
+            return self._section_via_block_id(block_id)
+
+    def next_section(self, section_id):
+        section_ids = tuple(self._sections_by_id.keys())
+        next_section_id = section_ids[section_ids.index(section_id) + 1]
+        return self.section(next_section_id)
+
+    def _section_via_block_id(self, block_id):
+        block = self.block(block_id)
+
+        if block:
+            try:
+                section_id = self.group(block['parent_id'])['parent_id']
+            except TypeError:
+                # Handle lists - they are nested one more level
+                section_id = self.group(self.block(block['parent_id'])['parent_id'])[
+                    'parent_id'
+                ]
+
+            return self.section(section_id)
+
     def groups(self):
         return self._groups_by_id.values()
 
-    @property
+    def group(self, group_id=None, block_id=None):
+        if group_id:
+            return self._groups_by_id.get(group_id)
+
+        if block_id:
+            return self._group_via_block(block_id)
+
+    def first_block_id_for_group(self, group_id):
+        group = self.group(group_id)
+        if group:
+            return group['blocks'][0]['id']
+
+    def _group_via_block(self, block_id):
+        block = self.block(block_id)
+        if block['type'] in LIST_COLLECTOR_CHILDREN:
+            parent = self.block(block['parent_id'])
+            return self.group(parent['parent_id'])
+        return self.group(block['parent_id'])
+
     def blocks(self):
         return self._blocks_by_id.values()
 
-    @property
-    def answers(self):
-        return self._answers_by_id.values()
+    def block(self, block_id=None, answer_id=None):
+        if block_id:
+            return self._blocks_by_id.get(block_id)
 
-    def get_section(self, section_id):
-        return self._sections_by_id.get(section_id)
+        if answer_id:
+            return self._block_via_answer(answer_id)
 
-    def get_group(self, group_id):
-        return self._groups_by_id.get(group_id)
+    def _block_via_answer(self, answer_id):
+        answers = self.answers(answer_id)
+        # All matching questions / answers must be within the same block
+        questions = self.questions(answers[0]['parent_id'])
+        return self.block(block_id=questions[0]['parent_id'])
 
-    def get_block(self, block_id):
-        return self._blocks_by_id.get(block_id)
+    def answer_ids(self):
+        return self._answers_by_id.values()
 
-    def get_block_for_answer_id(self, answer_id):
-        answers = self.get_answers(answer_id)
-        # All matching questions / answers must be within the same block
-        questions = self.get_questions(answers[0]['parent_id'])
-        return self.get_block(questions[0]['parent_id'])
+    def answers(self, answer_id):
```

Comment:
The method name here doesn't imply that it's actually filtering the answers collection.